Kodi 19.1 does not compile against libfmt 8.0.0

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Kodi 19.1 does not compile against libmt 8.0.0.

Compilation fails with this output:
```
In file included from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:11:
/var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h: In instantiation of ‘static std::wstring StringUtils::Format(const wstring&, Args&& ...) [with Args = {const wchar_t*}; std::wstring = std::__cxx11::basic_string<wchar_t>]’:
/var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:221:55:   required from here
/var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h:98:32: error: no matching function for call to ‘format(const wstring&, const wchar_t*&)’
   98 |     auto result = ::fmt::format(fmt, EnumToInt(std::forward<Args>(args))...);
      |                   ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/fmt/format.h:44,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h:34,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:11:
/usr/include/fmt/core.h:2889:17: note: candidate: ‘std::string fmt::v7::format(fmt::v7::format_string<T ...>, T&& ...) [with T = {const wchar_t*&}; std::string = std::__cxx11::basic_string<char>; fmt::v7::format_string<T ...> = fmt::v7::basic_format_string<char, const wchar_t*&>]’
 2889 | FMT_INLINE auto format(format_string<T...> fmt, T&&... args) -> std::string {
      |                 ^~~~~~
/usr/include/fmt/core.h:2889:44: note:   no known conversion for argument 1 from ‘const wstring’ {aka ‘const std::__cxx11::basic_string<wchar_t>’} to ‘fmt::v7::format_string<const wchar_t*&>’ {aka ‘fmt::v7::basic_format_string<char, const wchar_t*&>’}
 2889 | FMT_INLINE auto format(format_string<T...> fmt, T&&... args) -> std::string {
      |                        ~~~~~~~~~~~~~~~~~~~~^~~
In file included from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h:34,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:11:
/usr/include/fmt/format.h:2781:13: note: candidate: ‘template<class Locale, class ... T, typename std::enable_if<fmt::v7::detail::is_locale<Locale>::value, int>::type <anonymous> > std::string fmt::v7::format(const Locale&, fmt::v7::format_string<T ...>, T&& ...)’
 2781 | inline auto format(const Locale& loc, format_string<T...> fmt, T&&... args)
      |             ^~~~~~
/usr/include/fmt/format.h:2781:13: note:   template argument deduction/substitution failed:
In file included from /usr/include/fmt/format.h:44,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h:34,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:11:
/usr/include/fmt/format.h:2780:11: error: no type named ‘type’ in ‘struct std::enable_if<false, int>’
 2780 |           FMT_ENABLE_IF(detail::is_locale<Locale>::value)>
      |           ^~~~~~~~~~~~~
```

Originally reported at https://bugs.gentoo.org/797619

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
Kodi should be able to compile against the version 8.0.0 of fmt


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
Compilation fails.


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->



### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
Steps to reproduce the behavior:
1. Install fmt 6.2.0
2. Compile Kodi



### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:

n/a, this is a compilation problem, not a runtime one.

### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->

n/a, this is a compilation problem, not a runtime one.

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->

n/a, this is a compilation problem, not a runtime one.

### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [X] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Gentoo/Linux
 - Kodi version: 19.1



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/xbmc/utils/StringUtils.h
+++ b/xbmc/utils/StringUtils.h
@@ -31,10 +31,13 @@
 #undef FMT_DEPRECATED
 #define FMT_DEPRECATED
 #endif
-#include <fmt/format.h>
-
 #include "XBDateTime.h"
 #include "utils/params_check_macros.h"
+
+#include <fmt/format.h>
+#if FMT_VERSION >= 80000
+#include <fmt/xchar.h>
+#endif
 
 /*! \brief  C-processor Token stringification
 

